log/logClient.go

```diff
@@ -100,17 +103,28 @@ type Feeder struct {
 }
 
 // NewClient Function
-func NewClient(server, msgPath, logPath, logFilter string, limit uint32) *Feeder {
+func NewClient(server string, o Options, c kubernetes.Interface) *Feeder {
 	fd := &Feeder{}
 
 	fd.Running = true
 
 	fd.server = server
 
-	fd.limit = limit
+	fd.limit = o.Limit
 
-	conn, err := grpc.Dial(fd.server, grpc.WithInsecure())
+	var creds credentials.TransportCredentials
+	if !o.Insecure {
+		tlsCreds, err := loadTLSCredentials(c, o)
+		if err != nil {
+			return nil
+		}
+		creds = tlsCreds
+	} else {
+		creds = insecure.NewCredentials()
+	}
+	conn, err := grpc.Dial(fd.server, grpc.WithTransportCredentials(creds))
```

Comment:
Can we try to fallback to secure if grpc.Dial fails because server rejected insecure connection?

Reply:
if server has tls enabled the fallback logic will not work.